src/standalone/UIKit/InputControls/DateInput.tsx

```diff
@@ -25,30 +75,97 @@ const DateInput = (props: DateInputProps & DatePickerProps) => {
 		},
 		[openInfo]
 	);
+	const handleTextFieldOnChange = useCallback(
+		(event: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => {
+			handleCursorChange(event);
+
+			if (!onChange) return;
 
+			const formatted = TypeDate.format(event.target.value);
+			setTextValue(formatted);
+
+			const date = new Date(formatted);
+			// If the fields of date are NaN the date is invalid, we should raise an error for this (onError prop)
+			if (formatted.includes("_") || isNaN(date.getTime())) {
+				setErrorHelperText(
+					ccI18n.t(
+						"backend-integration.model.types.renderers.date.labels.invalid-date"
+					)
+				);
+				if (!isError) setIsError(true);
+			} else {
+				setErrorHelperText("");
+				setIsError(false);
+			}
+			onChange(formatted.includes("_") || isNaN(date.getMonth()) ? null : date);
+		},
+		[isError, setIsError, setErrorHelperText, onChange, handleCursorChange]
+	);
+
+	const handlePickerChange = useCallback(
+		(newDate: MaterialUiPickersDate) =>
+			onChange(newDate ? newDate.toDate() : null),
+		[onChange]
+	);
+	const handlePickerError = useCallback(
+		(error: React.ReactNode) => {
+			if (error) {
+				setErrorHelperText(error as string);
+				setIsError(true);
+			} else {
+				if (!textValue.includes("_")) {
+					setIsError(false);
+					setErrorHelperText("");
+				}
+			}
+		},
+		[textValue, setIsError, setErrorHelperText]
+	);
 	return (
-		<LocalizedDatePicker
-			{...muiProps}
-			clearable
-			InputProps={{
-				classes: inputClasses,
-				endAdornment: (
-					<InputAdornment position="end">
-						{!muiProps.disabled && (
-							<IconButton>
-								<CalenderIcon color={"disabled"} />
-							</IconButton>
-						)}
-						{openInfo && (
-							<IconButton onClick={handleOpenInfo}>
-								<InfoIcon color={"disabled"} />
-							</IconButton>
-						)}
-					</InputAdornment>
-				),
-			}}
-			InputLabelProps={InputLabelConfig}
-		/>
+		<>
+			<TextFieldWithHelp
+				fullWidth={fullWidth}
+				placeholder={placeholder}
+				value={textValue}
+				error={isError}
+				helperText={errorHelperText}
+				onChange={handleTextFieldOnChange}
+				InputProps={{
+					classes: inputClasses,
+					endAdornment: (
+						<InputAdornment position="end">
+							{!muiProps.disabled && (
+								<IconButton onClick={() => setIsOpen(true)}>
+									<CalenderIcon color={"disabled"} />
+								</IconButton>
+							)}
+							{openInfo && (
+								<IconButton onClick={handleOpenInfo}>
+									<InfoIcon color={"disabled"} />
+								</IconButton>
+							)}
+						</InputAdornment>
+					),
+				}}
+				inputProps={{
+					ref: cursorInputRef,
+				}}
+				InputLabelProps={InputLabelConfig}
+			/>
+
+			<LocalizedDatePicker
+				{...muiProps}
+				value={value}
+				onChange={handlePickerChange}
+				clearable
+				open={isOpen}
+				onError={handlePickerError}
+				onOpen={() => setIsOpen(true)}
+				onClose={() => setIsOpen(false)}
+				TextFieldComponent={() => null}
```

Comment:
TextFieldComponent can be made a static variable outside of the component to avoid rerendering
`const renderTextFieldComponent = () => null;`